src/test/java/seedu/address/model/person/predicates/StudentNumberContainsKeywordsPredicateTest.java

```diff
@@ -0,0 +1,76 @@
+package seedu.address.model.person.predicates;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.testutil.PersonBuilder;
+
+public class StudentNumberContainsKeywordsPredicateTest {
+
+    @Test
+    public void equals() {
+        List<String> firstPredicateKeywordList = Collections.singletonList("first");
+        List<String> secondPredicateKeywordList = Arrays.asList("first", "second");
+
+        StudentNumberContainsKeywordsPredicate firstPredicate =
+                new StudentNumberContainsKeywordsPredicate(firstPredicateKeywordList);
+        StudentNumberContainsKeywordsPredicate secondPredicate =
+                new StudentNumberContainsKeywordsPredicate(secondPredicateKeywordList);
+
+        // same object -> returns true
+        assertTrue(firstPredicate.equals(firstPredicate));
+
+        // same values -> returns true
+        StudentNumberContainsKeywordsPredicate firstPredicateCopy =
+                new StudentNumberContainsKeywordsPredicate(firstPredicateKeywordList);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different phone number -> returns false
```

Comment:
Different Student Number

Reply:
yup hahah just changed this is my current iteration, to be pushed soon!